crates/goose-server/src/routes/status.rs

```diff
@@ -0,0 +1,46 @@
+use axum::body::Body;
+use axum::http::HeaderValue;
+use axum::response::IntoResponse;
+use axum::{extract::Path, http::StatusCode, routing::get, Router};
+use goose::session::generate_diagnostics;
+
+#[utoipa::path(get, path = "/status",
+    responses(
+        (status = 200, description = "ok", body = String),
+    )
+)]
+async fn status() -> String {
+    "ok".to_string()
+}
+
+#[utoipa::path(get, path = "/diagnostics/{session_id}",
+    responses(
+        (status = 200, description = "Diagnostics zip file", content_type = "application/zip", body = Vec<u8>),
+        (status = 500, description = "Failed to generate diagnostics"),
+    )
+)]
+async fn diagnostics(Path(session_id): Path<String>) -> impl IntoResponse {
+    match generate_diagnostics(&session_id).await {
+        Ok(zip_data) => {
+            let filename = format!("attachment; filename=\"diagnostics_{}.zip\"", session_id);
+            let headers = [
+                (
+                    axum::http::header::CONTENT_TYPE,
+                    HeaderValue::from_static("application/zip"),
+                ),
+                (
+                    axum::http::header::CONTENT_DISPOSITION,
+                    HeaderValue::from_str(&filename).unwrap(),
```

Comment:
you have an .unwrap here, I think it's highly unlikely to fail to make a header value out of your file name, but might as well map to a 500

Reply:
yeah, we should really get around automatically doing that at some point